### PR TITLE
CMOS-56: extend the container example for use with multiple clusters

### PR DESCRIPTION
**Pull request by redm2**

### CODE REVIEW

File: examples/containers/multi/helpers/driver.sh [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+                  || echo 'failed'" "SUCCESS: "
+
+                nodes+=(\""$node".local:9091\")
+                echo " - $node added"
+        done
+
+        echo "All nodes added successfully."
+
+        # Add cluster to CMOS' Prometheus JSON config file
+        local csv
```
Comment (patrick-stephens) [inline examples/containers/multi/helpers/driver.sh:210]: separate function to do this

File: examples/containers/multi/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+services:
+    cmos:
+
+        container_name: cmos
+        image: "${CMOS_IMAGE}"
+        build:
```
Comment (patrick-stephens) [inline examples/containers/multi/docker-compose.yml:6]: I'd say remove and switch to built outside.

File: Makefile [hunk 3]
```diff
@@ -89,7 +94,17 @@ example-kubernetes: container
 example-containers: container
 	examples/containers/run.sh
 
-examples: clean container example-kubernetes example-containers
+example-multi: container
+	examples/containers/multi/run.sh
+
+example-kubernetes-oss: container-oss
```
Comment (patrick-stephens) [inline Makefile:100]: Yeah just remove all these I think

File: Makefile [hunk 4]
```diff
@@ -15,9 +15,14 @@ GIT_REVISION := $(shell git rev-parse HEAD)
 # This is analogous to revisions in DEB and RPM archives.
 revision = $(if $(REVISION),$(REVISION),)
 
-.PHONY: all build lint container container-oss config-svc-build config-svc-container config-svc-test-unit container-public container-lint container-scan dist test-dist container-clean clean examples example-containers test test-kubernetes test-native test-containers test-unit docs-license-analysis
+.PHONY: all build clean config-svc-lint container container-clean container-lint container-oss \
+		container-public container-scan dist docs docs-generate-markdown docs-license-analysis \
+		docs-lint example-kubernetes example-containers example-multi example-kubernetes-oss \
+		example-containers-oss example-multi-oss lint test test-containers test-dist test-kubernetes \
+		test-native
```
Comment (patrick-stephens) [inline Makefile:22]: bit of a weird split, might be better to keep `test-` together, etc.

**Review by patrick-stephens — CHANGES_REQUESTED** [review]
I think it's pretty much there apart from the large function one.

File: examples/containers/multi/run.sh [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+# Create $NUM_NODES containers running Couchbase Server $VERSION and the exporter
+start_new_nodes "$NUM_NODES" "cbs_server_exp"
+
+# Initialise and partition nodes as evenly as possible into $NUM_CLUSTERS clusters, register them with CBMM
+# and if $LOAD=true throw a light (non-zero) load at the cluster to simulate use using cbpillowfight
+configure_servers "$NUM_NODES" "$NUM_CLUSTERS" "$SERVER_USER" "$SERVER_PWD" "$NODE_RAM" "$LOAD" "$OSS_FLAG" 
+
+echo "All done. Go to: http://localhost:8080."
+
+if $OSS_FLAG; then
```
[the remark this answers is not in the document]
Comment (redm2) [reply examples/containers/multi/run.sh:107]: Fair enough yeah, it was a last minute bolt-on

File: Makefile (hunk 4 above)
Comment (redm2) [reply Makefile:22]: Yeah just did it alphabetically but the linebreaks could use some work

Comment (patrick-stephens) [reply Makefile:22]: Yeah probably easier to keep things together by "function" then most generic first before rest by alphabetical

`test test-containers test-dist test-native`

Comment (patrick-stephens) [reply Makefile:22]: Very much a nit pick though

**Review by patrick-stephens — APPROVED** [review]

**Pull request merged**